migrate/v0_15/migrate_test.go

```diff
@@ -99,3 +103,89 @@ func TestIncentive(t *testing.T) {
+	cdc.MustUnmarshalJSON(bz, &genesisState)
+
+	migratedGenesisState := Auth(genesisState, GenesisTime)
+
+	assert.Equal(t, genesisState.Params, migratedGenesisState.Params, "expected auth parameters to not change")
+}
+
+func TestAuth_AccountConversion(t *testing.T) {
+	bz, err := ioutil.ReadFile(filepath.Join("testdata", "kava-7-test-auth-state.json"))
+	require.NoError(t, err)
+
+	var genesisState auth.GenesisState
+	cdc := app.MakeCodec()
+	cdc.MustUnmarshalJSON(bz, &genesisState)
+
+	migratedGenesisState := Auth(genesisState, GenesisTime)
+	require.Equal(t, len(genesisState.Accounts), len(migratedGenesisState.Accounts), "expected the number of accounts after migration to be equal")
+
+	for i, acc := range migratedGenesisState.Accounts {
+		oldAcc := genesisState.Accounts[i]
+
+		// total owned coins does not change
+		require.Equal(t, oldAcc.GetCoins(), acc.GetCoins(), "expected base coins to not change")
+
+		// ensure spenable coins at genesis time is equal
+		require.Equal(t, oldAcc.SpendableCoins(GenesisTime), acc.SpendableCoins(GenesisTime), "expected spendable coins to not chain")
+		// check 30 days
+		futureDate := GenesisTime.Add(30 * 24 * time.Hour)
+		require.Equal(t, oldAcc.SpendableCoins(futureDate), acc.SpendableCoins(futureDate), "expected spendable coins to not chain")
+		// check 90 days
+		futureDate = GenesisTime.Add(90 * 24 * time.Hour)
+		require.Equal(t, oldAcc.SpendableCoins(futureDate), acc.SpendableCoins(futureDate), "expected spendable coins to not chain")
+		// check 180 days
+		futureDate = GenesisTime.Add(180 * 24 * time.Hour)
+		require.Equal(t, oldAcc.SpendableCoins(futureDate), acc.SpendableCoins(futureDate), "expected spendable coins to not chain")
+		// check 365 days
+		futureDate = GenesisTime.Add(365 * 24 * time.Hour)
+		require.Equal(t, oldAcc.SpendableCoins(futureDate), acc.SpendableCoins(futureDate), "expected spendable coins to not chain")
+
+		if vacc, ok := acc.(vesting.PeriodicVestingAccount); ok {
+			// old account must be a periodic vesting account
+			oldVacc, ok := oldAcc.(vesting.PeriodicVestingAccount)
+			require.True(t, ok)
+
+			// total delegated coins must match
+			oldTotalDelegated := oldVacc.DelegatedFree.Add(oldVacc.DelegatedVesting...)
+			newTotalDelegated := vacc.DelegatedFree.Add(vacc.DelegatedVesting...)
+			require.Equal(t, oldTotalDelegated, newTotalDelegated, "expected total amount of tracked delegations to not change")
+
+			// delegated vesting must be less or equal to original vesting
+			require.True(t, vacc.DelegatedVesting.IsAllLTE(vacc.OriginalVesting), "expected delegated vesting to be less or equal to original vesting")
+
+			// vested coins must be nil for the new account
+			require.Equal(t, sdk.Coins(nil), vacc.GetVestedCoins(GenesisTime), "expected no vested coins at genesis time")
+
+			// vesting coins must not be nil
+			require.NotEqual(t, sdk.Coins(nil), vacc.GetVestingCoins(GenesisTime), "expected vesting coins to be greater than 0")
```

Comment:
could maybe add an assertion here than the number of vesting periods on the new vesting account is less than or equal to the number of vesting periods on the old vesting account.